components/org.wso2.carbon.identity.oidc.session/src/main/java/org/wso2/carbon/identity/oidc/session/DefaultOIDCSessionStateManager.java

```diff
@@ -81,6 +84,64 @@ public Cookie addOPBrowserStateCookie(HttpServletResponse response) {
         return cookie;
     }
 
+    /**
+     * Adds the browser state cookie with tenant qualified path to the response.
+     *
+     * @param response
+     * @param request
+     * @param tenantDomain
+     * @return Cookie
+     */
+    @Override
+    public Cookie addOPBrowserStateCookie(HttpServletResponse response, HttpServletRequest request,
+                                          String tenantDomain) {
+
+        ServletCookie cookie;
+        if (IdentityTenantUtil.isTenantedSessionsEnabled() && tenantDomain != null) {
+            // Invalidate the old opbs cookies which haven't tenanted paths.
+            removeOPBrowserStateCookiesInRoot(request, response);
+
+            cookie = new ServletCookie(OIDCSessionConstants.OPBS_COOKIE_ID, UUID.randomUUID().toString() +
+                    OIDCSessionConstants.TENANT_QUALIFIED_OPBS_COOKIE_SUFFIX);
+            cookie.setSecure(true);
+            cookie.setPath(FrameworkConstants.TENANT_CONTEXT_PREFIX + tenantDomain + "/");
+        } else {
+            cookie = new ServletCookie(OIDCSessionConstants.OPBS_COOKIE_ID, UUID.randomUUID().toString());
+            cookie.setSecure(true);
+            cookie.setPath("/");
+        }
+        cookie.setSameSite(SameSiteCookie.NONE);
+        response.addCookie(cookie);
+        return cookie;
+    }
+
+    /**
+     * Invalidate the old opbs cookies which haven't tenanted paths.
+     *
+     * @param request
+     * @param response
+     */
+    private static void removeOPBrowserStateCookiesInRoot (HttpServletRequest request, HttpServletResponse response) {
+
+        Cookie[] cookies = request.getCookies();
+        if (cookies != null) {
```

Comment:
Can we simplify this if condition?

if(cookies = null){
log.debug(..);
return;
}

// have logic to remove cookie